File: DesignPatternsExercises/java/StrategyDesignPattern/src/TSVExtraction.java
class TSVExtraction implements Extraction {

	public void extract() {
		// __________(10)____________
		open();

		// __________(11)____________
		System.out.println("I will extract by separating row by tabs.");

		// __________(12)____________
		close();
	}

	public void open() {
		System.out.println("Opening TSV file");
	}

	public void close() {
		System.out.println("Closing TSV file");
	}
}

File: DesignPatternsExercises/ruby/strategy_design_pattern/exercise/tsv_extraction.rb
require_relative 'extraction'

class TSVExtraction < Extraction
  def extract
    __________(8)__________
    puts 'TSV Data type is getting extracted by separating row by tabs...'
    __________(9)__________
  end

  def open
    puts 'Opening TSV file...'
  end

  def close
    puts 'Closing TSV file...'
  end
end
